app/services/renew_book.rb

```diff
@@ -0,0 +1,24 @@
+class RenewBook
+  def self.process(checkout:, user:)
+    new(checkout: checkout, user: user).process
+  end
+
+  def initialize(checkout:, user:)
+    @checkout = checkout
+    @user = user
+  end
+
+  def process
+    book = Book.find(@checkout.book_id)
+    due_date = @checkout.due_on + Checkout::CHECKOUT_PERIOD_IN_DAYS.days
+    book_fine = @checkout.calculate_fine
```

Comment:
I'd put this, `book` and `due_date` in private methods. It will make it easier to see exactly what this method does